infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/OpenShiftClientFactory.java

```diff
@@ -52,7 +59,7 @@ public OpenShiftClientFactory(
     if (doTrustCerts != null) {
       configBuilder.withTrustCerts(doTrustCerts);
     }
-    config = configBuilder.build();
+    this.client = new DefaultOpenShiftClient(configBuilder.build());
```

Comment:
Consider adding a decorator that would eliminate the risk of calling close on the client.